lib/bap_service/bap_service.mli

```diff
@@ -0,0 +1,69 @@
+
+(** Bap-service library.
+
+    {1 Overview}
+
+    The library establishes simple relationships between different
+    sides of computation.
+
+    The [service] is an abstraction of resources on which a
+    computation may depend.
+
+    Services are provided by service [providers]. The same service
+    could be provided by several providers. Only a service provider
+    knows how the provided service depends on the environment.
+
+    A [product] is a result of a particular comptutation, that depends
+    of computational environment. *)
+
+open Core_kernel.Std
+open Bap_future.Std
+open Regular.Std
+
+type service
+type provider
+type product [@@deriving bin_io, compare, sexp]
+
+type void
+type literal = (void,void,void) format
+
+module Service : sig
+  type t = service
+
+  (** [declare ~desc uuid name] creates a service from description
+      [desc], unique id [uuid] and [name] *)
```

Comment:
this documentation says nothing. It basically reiterates the signature. 
It doesn't tell a user that the [uuid] must be statically unique and will be used to distinguish services globally, that `name` is a short and descriptive name, and description provides a short description of the provided service. 
It also doesn't explain what effects this declaration has. 

Again, imagine that you're a user and you're trying to understand how to use this function.